pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -206,74 +207,55 @@ def circuit(params):
         Array(0.94301294, dtype=float64)
         >>> jax.grad(circuit)(params)
         [Array(0.59484969, dtype=float64)]
-
     """
     if isinstance(wires, int):
         wires = [wires]
-    trivial_detuning = not callable(detuning) and qml.math.isclose(detuning, 0.0)
 
     if not callable(amplitude) and qml.math.isclose(amplitude, 0.0):
-        if trivial_detuning:
-            raise ValueError(
-                f"Expected non-zero value for at least one of either amplitude or detuning, but received amplitude={amplitude} and detuning={detuning}. All terms are zero."
-            )
-
-        coeffs = []
-        observables = []
+        raise ValueError(
+            f"Expected non-zero value or callable for amplitude, but received amplitude={amplitude}. All terms are zero."
+        )
 
     # TODO: use sigma+ and sigma- (not necessary as terms are the same, but for consistency)
-    # We compute the `coeffs` and `observables` of the electromagnetic field
-    else:
-        # We compute the `coeffs` and `observables` of the EM field
-        coeffs = [
-            amplitude_and_phase(qml.math.cos, amplitude, phase),
-            amplitude_and_phase(qml.math.sin, amplitude, phase),
-        ]
-
-        drive_x_term = 0.5 * sum(qml.PauliX(wire) for wire in wires)
-        drive_y_term = -0.5 * sum(qml.PauliY(wire) for wire in wires)
+    # We compute the `coeffs` and `observables` of the EM field
+    coeffs = [
+        amplitude_and_phase(qml.math.cos, amplitude, phase),
+        amplitude_and_phase(qml.math.sin, amplitude, phase),
+    ]
 
-        observables = [drive_x_term, drive_y_term]
+    drive_x_term = 0.5 * sum(qml.PauliX(wire) for wire in wires)
+    drive_y_term = -0.5 * sum(qml.PauliY(wire) for wire in wires)
 
-    if not trivial_detuning:
-        detuning_term = -1.0 * sum(qml.PauliZ(wire) for wire in wires)
-        coeffs.append(detuning)
-        observables.append(detuning_term)
+    observables = [drive_x_term, drive_y_term]
 
     # We convert the pulse data into a list of ``HardwarePulse`` objects
-    pulses = [HardwarePulse(amplitude, phase, detuning, wires)]
+    frequency = None
```

Comment:
I guess the idea is that this will come in a future PR? I'm hesitant to do that, because we currently have something working for the Rydberg Hamiltonian, and I'd like that to continue to be the case so that the tests etc can run from master for the plugin during AWS review. Merging this as-is would mean removing from master the option to have a Hamiltonian with detuning for the Rydberg devices.

Reply:
Ah yes we need to be able to pass something for ``frequency`` here when we wrap it for `transmon_drive`, good point 🤔

Reply:
So we need `drive` to be a `HardwareHamiltonian` due to the custom call ordering, but at the same time we want to wrap `drive` in `rydberg_drive` with an additional `detuning` (or `frequency`) term. So maybe we can just create the terms through drive but overwrite the `HardwareHamiltonian.pulses` attribute, or, more realistically, just instantiate a new `HardwareHamiltonian` with the correct `pulses` passed from the start. Both valid options. CC @mudit2812 as you're going to be working on creating the `rydberg_drive`.

Reply:
Actually, after a bit more consideration, I think it's a better idea to not specify `pulses` in `drive` at all if it's not going to be exposed to the user, and it's just as easy to specify it in `transmon_drive` and `rydberg_drive`. Both of those functions would otherwise overwrite `HardwareHamiltonian.pulses` anyway.